### DISCUSSION TITLE
Trapping interactive mesh slices using add_mesh_slice()

### DISCUSSION THREAD

**Discussion by danwbean** [pyvista/pyvista#2644] [Q&A]
I am working on a tool to compare cross-section between different stone projectile points.
I need to get a callback when the widget _add_mesh_slice_ is moved.  I will add the section created and saved in plotter.plane_sliced_meshes in a _different_ window.

![Slice_thru_ppt](https://user-images.githubusercontent.com/25962619/131582257-9e0c84c0-9e9c-4d4c-bc5d-d3e828217a9f.png)

Here is simplified toy code that demonstrates the problem:

import pyvista as pv

mesh = pv.Cone()
p = pv.Plotter()

p.window_size=(1000, 1000)
p.background_color = 'black'
p.add_mesh(mesh,  opacity=0.5)

p.add_mesh_slice(mesh, assign_to_axis='x', generate_triangles=False)

print('XZ number of points:', p.plane_sliced_meshes[0].number_of_points)       # how to catch this event every move

p.show()

How do I catch each move of the slicing plane?
For example:  Get the print statement trigger each move (rather than once as it does now)

I see the callback in the widget.py class, but do not know how to trap it and see the p.plane_sliced_meshes at the same time

**Comment by chautuankien** [comment]
I am also want to do similar thing. Dis you solve this problem?

**Comment by edabor** [comment]
@danwbean and @chautuankien Here is some code that might help you doing what you intend (I modified the one you provided).

The trick here is to use the `vtkAlgorithm` support in `plotter.add_mesh` (see https://docs.pyvista.org/version/stable/examples/99-advanced/plotting-algorithms.html#).

```python
import pyvista as pv
import vtk

mesh = pv.Cone()
p = pv.Plotter(shape=(1, 2))
p.window_size = (1000, 1000)
p.set_background('black', all_renderers=True)

cutter = vtk.vtkCutter()
plane = vtk.vtkPlane()
cutter.SetInputData(mesh)
cutter.SetCutFunction(plane)

plane.SetOrigin(mesh.center)
plane.SetNormal((1, 0, 0))

def callback(normal, origin):
    plane.SetOrigin(origin)
    plane.SetNormal(normal)
    cutter.Update()
    p.subplot(0, 1)
    p.camera.reset_clipping_range()

p.subplot(0, 1)
p.add_mesh(cutter)
p.view_yz()

p.subplot(0, 0)
p.add_mesh(mesh, opacity=0.5)
p.add_plane_widget(callback=callback, interaction_event="always", assign_to_axis='x')
p.show()
```
